src/spoke/Spoke.sol

```diff
@@ -299,42 +299,50 @@ abstract contract Spoke is ISpoke, Multicall, NoncesKeyed, AccessManagedUpgradea
   ) external onlyPositionManager(onBehalfOf) returns (uint256, uint256) {
     Reserve storage reserve = _getReserve(reserveId);
     UserPosition storage userPosition = _userPositions[onBehalfOf][reserveId];
+    PositionStatus storage positionStatus = _positionStatus[onBehalfOf];
     _validateRepay(reserve);
 
+    IHubBase.PremiumDelta memory premiumDelta;
+    uint256 drawnDebtRestored = amount;
```

Comment:
why?

Reply:
passing `amount` later in the method was creating StackTooDeep error.
I'll try to remove this later on when done with all logic changes.